### ISSUE TITLE
Set method can be safe

### ISSUE THREAD

**Issue by matklad** [sagebind/staticslot#4]
Leaking data and not calling destruction is explicitly allowed in Rust, so `set` does not need to be unsafe.

**Comment by sagebind** [comment]
True, I think that semantics changed. Back when staticslot was originally written, `Drop` types were not allowed in static variables because leaking used to be considered not allowed.
